src/inmanta/util.py

```diff
@@ -81,6 +81,11 @@ def ensure_directory_exist(directory: str, *subdirs: str) -> str:
     return directory
 
 
+def is_sub_dict(subdct: Dict[PrimitiveTypes, PrimitiveTypes], dct: Dict[PrimitiveTypes, PrimitiveTypes]):
+    matching_key_value_pairs = {k: v for k, v in subdct.items() if k in dct and dct[k] == v}
+    return len(matching_key_value_pairs) == len(subdct)
```

Comment:
minor comment, a potentially more efficient version would be

```suggestion
   return not any(k for k,v in subdct.items() if k not in dct or dct[k] != v)
```

Reply:
(no multi line suggestions yet it would seem)

Reply:
Should it be: return not any(True for k,v in subdct.items() if k not in dct or dct[k] != v) instead? Otherwise dicts with a boolean value as a key are not handled correctly.

Reply:
Wow, good catch